torch/distributed/rpc/__init__.py

```diff
@@ -43,14 +43,22 @@ def init_model_parallel(
             init_method(str): backend specific init arguments.
             num_send_recv_threads(int): Number of threads for send/recv work.
         """
+        # Rendezvous.
+        world_size = len(worker_name_to_id)
+        rendezvous_iterator = torch.distributed.rendezvous(
```

Comment:
I recall that we ran into an issue of rendezvous twice when using `ProcessGroupAgent` when doing it here. Wouldn't this hit that problem again?

Reply:
I do handle this by inserting a few process_group barrier to circumvent it. 
Putting barrier between failed rendezvous versions could avoid mis-match.

Reply:
What do you mean here? Mismatch between torch.distributed and torch.distributed.rpc?

Reply:
Mismatch between multiple calls to `rendezvous`, an issue described in #28552.